cfme/utils/log_validator.py

```diff
@@ -101,3 +112,25 @@ def validate():
             except Failed:
                 return False
         wait_for(validate, delay=delay, num_sec=num_sec, message=message, **kwargs)
+
+    def wait_for_expected_patterns_match_count(
+        self,
+        expected_match,
+        timeout=180,
+        delay=5,
+        message="Check for expected match count",
+        **kwargs
+    ):
+        """ wait for expected match count found in respective log file.
+
+        Args:
+            expected_match (dir): expected patterns match count
```

Comment:
`dir` for expected match?  This docblock needs better descriptions here, as well as examples below.

Something like `'matches_and_counts (dict): patterns to match as dict keys, and their expected count as values'`

This could easily be a list of tuples as well, which might be a better way to structure such data.

Ultimately, since this is expecting the exact same data format as the LogValidator property, its a convenience method that I'm not sure adds a lot of value.

Did you consider actually changing how LogValidator.matched_patterns actually is stored, so that LogValidator.validate_logs is inherently validating the match and count?

It looks to me like we could default the `count` of a `matched_patterns` item to None (preserving current uses of LogValidator), but otherwise accept tuples of `(pattern, count)` in the `matched_patterns` list.  When the `count` is the default, `None`, we consider any number of matches a passing validation.

This way, the behavior of validation including the count would be inherent to a validation call, and the use case you've implemented here would simply be covered by `wait_for_log_validation`.

Then instead of separating

Reply:
wait for matches not a part of LogValidator now. User will implement as per need.

Reply:
Discussed scope and needs offline, keeping this PR's scope reduced and will introduce more changes to LogValidator later.